field/TinyLang/Field/Evaluator.hs

```diff
@@ -211,15 +212,11 @@ denoteExpr env = fmap denoteUniConst . evalExprUni env
 
 normStatement
     :: (MonadEvalError f m, Eq f, Field f, AsInteger f)
-    => Env (SomeUniConst f) -> Statement f -> Expr f a -> m (Expr f a)
-normStatement env (ELet (UniVar uni var) def) rest = do
-    defN <- normExpr env def
-    case defN of
-        EConst uniConst -> normExpr (insertVar var (Some uniConst) env) rest
-        _             -> EStatement (ELet (UniVar uni var) defN) <$> normExpr env rest
-normStatement env (EAssert expr) rest = do
-    exprN <- normExpr env expr
-    EStatement (EAssert exprN) <$> normExpr env rest
+    => Env (SomeUniConst f) -> Statement f -> m (Statement f)
+normStatement env (ELet uniVar def) = do
+    ELet uniVar <$> normExpr env def
```

Comment:
That changes the behavior. Previously we would inline things like `let x = 3`, but now we don't. This is needed for shrinking, so we have to fix that.